src/ucm/util/sys.c

```diff
@@ -349,3 +350,19 @@ char *ucm_concat_path(char *buffer, size_t max, const char *dir, const char *fil
 
     return buffer;
 }
+
+ucs_status_t ucm_mem_attr_get(const void *address, size_t length,
+                              ucs_memory_attr_t *mem_attr)
+{
+    ucm_event_installer_t *event_installer;
+
+    ucs_list_for_each(event_installer, &ucm_event_installer_list, list) {
+        if (NULL == event_installer->get_mem_attr) continue;
+        ucs_status_t status = event_installer->get_mem_attr(address, length,
+                                                            mem_attr);
+        if (status != UCS_ERR_INVALID_ADDR) return status;
```

Comment:
if `status` is NO_RESOURCE (say because both rocm and cuda event installers are present and rocm is seen first), then we would end up return returning NO_RESOURCE to mem_attr_get when in fact if we progressed and looked through cuda installer you may have obtained the right attribute.

Reply:
In other words, you may have to exhaust all installers before returning not UCS_OK status.